lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -598,7 +603,7 @@ public function __sleep(): array
      */
     public function slice($offset, $length = null): array
     {
-        if (! $this->initialized && ! $this->isDirty && $this->association['fetch'] === ClassMetadata::FETCH_EXTRA_LAZY) {
+        if (! $this->initialized && ! $this->isDirty && $this->getMapping()['fetch'] === ClassMetadata::FETCH_EXTRA_LAZY) {
```

Comment:
Same here

Reply:
Do I replace occurences when null is checked before like :

```php
    private function changed(): void
    {
        if ($this->isDirty) {
            return;
        }

        $this->isDirty = true;

        if (
            $this->association !== null &&
            $this->getMapping()['isOwningSide'] &&
            $this->getMapping()['type'] === ClassMetadata::MANY_TO_MANY &&
            $this->owner &&
            $this->em !== null &&
            $this->em->getClassMetadata(get_class($this->owner))->isChangeTrackingNotify()
        ) {
            $this->getUnitOfWork()->scheduleForDirtyCheck($this->owner);
        }
    }
```

It can maybe result side effects.